CRM/Custom/Form/Field.php

```diff
@@ -318,21 +318,26 @@ public function buildQuickForm() {
     if ($this->_action == CRM_Core_Action::UPDATE) {
       $this->freeze('data_type');
     }
-    $includeFieldIds = NULL;
+
+    $optionGroupParams = array(
+      'is_reserved' => 0,
+      'is_active' => 1,
+      'options' => array('limit' => 0, 'sort' => "title ASC"),
+    );
     if ($this->_action == CRM_Core_Action::UPDATE) {
-      $includeFieldIds = $this->_values['id'];
+      $optionGroupParams['id'] = $this->_values['id'];
     }
-    $optionGroups = CRM_Core_BAO_CustomField::customOptionGroup($includeFieldIds);
-    $emptyOptGroup = FALSE;
-    if (empty($optionGroups)) {
-      $emptyOptGroup = TRUE;
-      $optionTypes = array('1' => ts('Create a new set of options'));
-    }
-    else {
-      $optionTypes = array(
-        '1' => ts('Create a new set of options'),
-        '2' => ts('Reuse an existing set'),
-      );
+
+    // OptionGroup selection
+    $optionTypes = array('1' => ts('Create a new set of options'));
+    // Get all custom (is_reserved=0) option groups
+    $optionGroupMetadata = civicrm_api3('OptionGroup', 'get', $optionGroupParams);
+    if (!empty($optionGroupMetadata['values'])) {
+      $emptyOptGroup = FALSE;
+      foreach ($optionGroupMetadata['values'] as $id => $metadata) {
+        $optionGroups[$id] = $metadata['title'];
+      }
```

Comment:
Consider using `CRM_Utils_Array::collect()` instead of a loop.

Reply:
Nice! I didn't know about that function.